### ISSUE TITLE
npm package boilerplate setup

### ISSUE THREAD

**Issue by thisismydesign** [c-hive/team-contribution-calendar#32]
Ran into a few things when setting up another npm project which could be implemented here as well:
- .editorconfig
- vscode workspace settings for prettier autoformatting
- prettier and related eslint dev dependencies
- .eslintrc.js rules and prettier plugin
- cross-env

See:
- https://github.com/c-hive/x-rate-limiter/commit/5dd80a5a07a764086d354521afee333a71b0380f
- https://github.com/c-hive/x-rate-limiter/commit/f13fa92515e470490f20147205b11799c6d8c179

And a question: why did you need to explicitly require a file in the test script? `--require ./src/utils/TestUtils/mockNonJsResources.js`

**Comment by gomorizsolt** [comment]
Left the related SO answer in `mockNonJsResources.js`. The file is included during the runtime in order to mock non-js resource files(`.svg` in our case).

**Comment by thisismydesign** [comment]
`require.extensions` is deprecated and my other problem is that the test script should not depend on internal file structure. Is it possible to move it to the top of the test file? Or what about this alternative? https://github.com/bahmutov/node-hook

**Comment by gomorizsolt** [comment]
Yes, it's deprecated, but at least isn't locked.

Btw, found a pure solution by using [**ignore-styles**](https://www.npmjs.com/package/ignore-styles). The package supports mocking a wide range of file extensions, therefore the `mockNonJsResources.js` file could be removed(#33).

**Example**

```
"test:only": "mocha src/ --require ignore-styles --require @babel/register --require @babel/polyfill --recursive"
```

As a result, it's not dependent on any kind of internal file - fits the requirement you mentioned in the comment above.

**Comment by gomorizsolt** [comment]
Resolved by #33.
